src/main/java/com/google/cloud/security/scanner/actions/extractors/FileToState.java

```diff
@@ -54,13 +55,19 @@ public void processElement(ProcessContext processContext) throws IllegalArgument
 
       GCPProject project = new GCPProject(filePath.get(1), filePath.get(0));
       Gson gson = new Gson();
-      List<PolicyBinding> bindings = Arrays
-          .asList(gson.fromJson(fileContent, PolicyBinding[].class));
-      GCPResourceState policy = new GCPResourcePolicy(project, bindings);
-      processContext.output(KV.of((GCPResource) project, policy));
+      try {
+        List<PolicyBinding> bindings = Arrays.asList(
+            gson.fromJson(fileContent, PolicyBinding[].class));
+        GCPResourceState policy = new GCPResourcePolicy(project, bindings);
+        processContext.output(KV.of((GCPResource) project, policy));
+      } catch (JsonSyntaxException jse) {
+        throw new IllegalArgumentException("Policy file has invalid json. "
+            + "Check \"" + filePath.get(0) + "/" + filePath.get(1) + "/"
+            + filePath.get(2) + "\"\nFile content is: \n" + fileContent, jse);
+      }
     }
     else {
-      throw new IllegalArgumentException("Malformed input to FileToState.");
+      throw new IllegalArgumentException("Malformed input to FileToState. Filepath: " + filePath);
```

Comment:
Does the + operator really work with types String, List<String>? I would try to use String.format if possible, but please use your discretion.